lib/java/logging-service-telemetry/src/main/java/org/enso/logging/service/telemetry/TelemetryAppenderImpl.java

```diff
@@ -26,33 +22,35 @@
 @org.openide.util.lookup.ServiceProvider(service = TelemetryAppender.class)
 public final class TelemetryAppenderImpl extends TelemetryAppender {
   private static final String CREDENTIALS_FILE_ENV = "ENSO_CLOUD_CREDENTIALS_FILE";
-
-  /**
-   * We still want to limit the batch size to some reasonable number - sending too many logs in one
-   * request could also be problematic.
-   */
-  private static final int MAX_BATCH_SIZE = 100;
-
-  private static final int MAX_RETRIES = 5;
-  private static final Logger LOGGER = LoggerFactory.getLogger(TelemetryAppender.class.getName());
-
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(TelemetryAppenderImpl.class.getName());
   private Credentials credentials;
-  private final LogJobsQueue logQueue = new LogJobsQueue();
-
-  private HttpClient httpClient;
-
-  /**
-   * Set to true once an error is encountered when sending a request. In such case, it is most
-   * probably that no further requests will be successful. This flag is used to terminate the
-   * background thread.
-   */
-  private boolean requestSendingFailure;
+  private boolean credentialsParseFailure;
+  private LogJobsProcessor logJobsProcessor;
 
   @Override
   protected void append(ILoggingEvent eventObject) {
     enqueueJob(eventObject);
   }
 
+  private static Credentials readCredentials() {
+    var credentialsFile = credentialsFile();
+    if (!credentialsFile.toFile().exists()) {
+      LOGGER.warn("Credentials file not found at '{}'. Will not send telemetry", credentialsFile);
+      return null;
+    }
+    Credentials credentials;
+    try {
+      credentials = parseCredentials(credentialsFile);
+    } catch (IOException e) {
+      LOGGER.warn(
+          "Failed to parse credentials from '{}'. Will not send telemetry", credentialsFile);
+      return null;
+    }
+    LOGGER.debug("Credentials read from '{}': {}", credentialsFile, credentials);
```

Comment:
`credentials` should probably not be logged.

Reply:
[Credentials.toString](https://github.com/enso-org/enso/blob/4c8103cc4f2b8ccdbdb7771f7be966e47c1b941a/lib/java/logging-service-telemetry/src/main/scala/org/enso/logging/service/telemetry/Credentials.scala#L21-L27) are "masked"